driver-mapping/src/test/java/com/datastax/driver/mapping/MapperKeyspaceTest.java

```diff
@@ -0,0 +1,273 @@
+            Mapper<UserT> mapper = manager.mapper(userClass, mapperKeyspace);
+
+            mapper.save(sampleUser);
+            assertThat(expectedKeyspace).isNotNull();
+            Row row = session().execute(String.format("SELECT * FROM %s.user WHERE name = ?", expectedKeyspace), sampleUser.getName()).one();
+            assertThat(row).isNotNull();
+
+            UserT loadedUser = mapper.get(sampleUser.getName());
+            assertThat(loadedUser.getName()).isEqualTo(sampleUser.getName());
+            assertThat(loadedUser.getAddress().getStreet()).isEqualTo(sampleUser.getAddress().getStreet());
+
+            mapper.delete(sampleUser);
+            row = session().execute(String.format("SELECT * FROM %s.user WHERE name = ?", expectedKeyspace), sampleUser.getName()).one();
+            assertThat(row).isNull();
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public static class Address {
+        private String street;
+
+        public Address() {
+        }
+
+        public Address(String street) {
+            this.street = street;
+        }
+
+        public String getStreet() {
+            return street;
+        }
+
+        public void setStreet(String street) {
+            this.street = street;
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @UDT(keyspace = "MapperKeyspaceTest", name = "address")
+    public static class AddressWithKeyspace extends Address {
+        public AddressWithKeyspace() {
+        }
+
+        public AddressWithKeyspace(String street) {
+            super(street);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @UDT(name = "address")
+    public static class AddressWithoutKeyspace extends Address {
+        public AddressWithoutKeyspace() {
+        }
+
+        public AddressWithoutKeyspace(String street) {
+            super(street);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public static class User<AddressT extends Address> {
+        @PartitionKey
+        private String name;
+        private AddressT address;
+
+        public User() {
+        }
+
+        public User(String name, AddressT address) {
+            this.name = name;
+            this.address = address;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public AddressT getAddress() {
+            return address;
+        }
+
+        public void setAddress(AddressT address) {
+            this.address = address;
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Table(keyspace = "MapperKeyspaceTest", name = "user")
+    public static class UserWithKeyspace extends User<AddressWithKeyspace> {
+        public UserWithKeyspace() {
+        }
+
+        public UserWithKeyspace(String name, AddressWithKeyspace address) {
+            super(name, address);
+        }
+
+        @Override
+        public AddressWithKeyspace getAddress() {
```

Comment:
This getter has to be overridden, otherwise the mapper tries to map to `AddressT`. That's an edge case but I'll open an other ticket.

Reply:
FYI: https://datastax-oss.atlassian.net/browse/JAVA-1719